src/main/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQueryEnvironment.java

```diff
@@ -322,15 +326,13 @@ private CqueryNode getConfiguredTarget(
   @Nullable
   protected CqueryNode getValueFromKey(SkyKey key) throws InterruptedException {
     SkyValue value = getConfiguredTargetValue(key);
-    if (value == null) {
-      return null;
-    } else if (value instanceof ConfiguredTargetValue configuredTargetValue) {
-      return configuredTargetValue.getConfiguredTarget();
-    } else if (value instanceof AspectValue && key instanceof AspectKey aspectValue) {
-      return aspectValue;
-    } else {
-      throw new IllegalStateException("unknown value type for CqueryNode");
-    }
+    return switch (value) {
+      case ConfiguredTargetValue configuredTargetValue ->
+          configuredTargetValue.getConfiguredTarget();
+      case AspectValue ignored when key instanceof AspectKey aspectKey -> aspectKey;
```

Comment:
`ignored when` is part of Java switch expressions?

I find it a bit hard to parse this line, especially since the function is returning a value and this returns what looks like a key (even though the logic checks the key is a value). Is there any way to be a bit more clear?

Reply:
I also find this surprising, but this is a pure refactoring - I found the preexisting logic more confusing. Happy to change this though, I just don't know how without potentially breaking something.

Reply:
Done